### ISSUE TITLE
Lint/SafeNavigationConsistency precedence problem

### ISSUE THREAD

**Issue by Am1ne77** [rubocop/rubocop#13402]
Hello!

I believe that there might be an operator precedence problem with Lint/SafeNavigationConsistency.

```
 $ cat tmp/rubocop.rb
```
```ruby
 # frozen_string_literal: true

 def f(code)
   code&.foo? && code.bar || code&.baz
 end

 def g(code)
   (code&.foo? && code.bar) || code&.baz
 end
```

```
 $ bundle exec rubocop tmp/rubocop.rb
 Inspecting 1 file
 E

 Offenses:

 tmp/rubocop.rb:4:33: E: [Correctable] Lint/SafeNavigationConsistency: Use . instead of unnecessary &..
   code&.foo? && code.bar || code&.baz
                                ^^

 1 file inspected, 1 offense detected, 1 offense autocorrectable
```

--------

## Expected behavior

`code&.baz` is not an offense since `&&`  might evaluate to false because `code` was `nil`.
We get the correct behavior (no offense detected) if we force the precedence using parentheses.

## Actual behavior

Offense detected for statement after `||` operator.

## Steps to reproduce the problem

See above.

## RuboCop version

```
$ [bundle exec] rubocop -V
1.68.0 (using Parser 3.3.5.0, rubocop-ast 1.33.0, analyzing as Ruby 3.3, running on ruby 3.3.4) [arm64-darwin23]
  - rubocop-performance 1.22.1
```

**Comment by koic** [comment]
Since `code.bar` confirmed that `code` is not null, the safe navigation in `code&.baz` is unnecessary. In other words, this is not a false positive, but rather a false negative due to an inability to handle parentheses correctly. I will take a closer look at this issue.

**Comment by Am1ne77** [comment]
> Since `code.bar` confirmed that `code` is not null, the safe navigation in `code&.baz` is unnecessary. In other words, this is not a false positive, but rather a false negative due to an inability to handle parentheses correctly. I will take a closer look at this issue.

I disagree. In Ruby, the `&&` operator has a higher precedence than the `||` operator ([Ruby precedence](https://docs.ruby-lang.org/en/3.3/syntax/precedence_rdoc.html)), meaning that the function `g` in the example above represents the correct precedence of the `f` function. In the event that `code` is `nil` in `f`, we will first evaluate `code&.foo?`, since `code` is `nil`, the expression will return false. The `code.bar` expression will not be evaluated because of `&&`'s lazy evaluation, meaning that the next expression we will evaluate will be `code&.baz`. Since `code` is `nil`, we would get an error if we had `code.baz`. Here's a small example to illustrate that:

```ruby
class A
  def foo
    false
  end

  def bar
    false
  end

  def baz
    false
  end
end

code = nil
code&.foo && code.bar || code.baz
```

The above program produces:
```
undefined method `baz' for nil:NilClass (NoMethodError)
```
The program is valid however with `code&.baz`.

**Comment by koic** [comment]
Ah, you’re right. I'll review this with that in mind. Thank you for the additional information.
